typing_inspect.py

```diff
@@ -75,7 +75,7 @@ def is_generic_type(tp):
     """
     if NEW_TYPING:
         return (isinstance(tp, type) and issubclass(tp, Generic) or
-                isinstance(tp, _GenericAlias) and
+                (isinstance(tp, _GenericAlias) or isinstance(tp, _SpecialGenericAlias)) and
```

Comment:
Just a note: `isinstance` accepts tuples of types and `isinstance(tp, (_GenericAlias, _SpecialGenericAlias))` could be easier to read.